bundles/org.openhab.core.config.discovery.addon.process/src/main/java/org/openhab/core/config/discovery/addon/process/ProcessAddonFinder.java

```diff
@@ -92,6 +92,7 @@ public Set<AddonInfo> getSuggestedAddons() {
 
         try {
             processInfos = ProcessHandle.allProcesses().map(process -> new ProcessInfo(process.info()))
+                    .filter(info -> (info.command != null) || (info.commandLine != null))
```

Comment:
This looks much simpler, but still leaves the problem of duplicates.
My Windows system gave me 75 entries with the implementation based on the record, and about 200 with this implementation. A lot of duplicate entries for firefox etc.
A ```Set``` will contain unique objects - if the compare/equals functions are implemented properly. This worked before on the set of strings. It seems not to work properly unless I implement ```equals``` and ```hashcode```  for ```ProcessInfo``` like this:
```
        @Override
        public boolean equals(@Nullable Object other) {
            if (other instanceof ProcessInfo pi) {
                return (Objects.equals(command, pi.command) && Objects.equals(commandLine, pi.commandLine));
            }
            return false;
        }

        @Override
        public int hashCode() {
            return Objects.hashCode(command) + Objects.hashCode(commandLine);
        }
```

Reply:
@holgerfriedrich what you call 'duplicates' are not really duplicates. Each entry represents a distinct process. It may be that a single executable has started many processes (most likely each process having a different command line; which on Windows is nulled out). I can imagine this case as you say with a browser having many contexts, or a service runner starting many services, or a script runner running many scripts. Obviously implementing an 'equals' method would allow the set to be smaller, but I am not sure that it is really proper to consider different the ProcessInfo for such processes to be 'equal'. => @mherwege WDYT?

Reply:
From my perspective, any duplicate in the list - where ever it originates from - does not add any value to the matcher. Duplicates in this case mean that both ```command``` and ```commandLine``` of two entries are the same.
Why should the matcher do a regex matching for the same input twice? This is why I used ```Set``` in the original implementation, which implicitly removed duplicate entries.

Reply:
Yes. OK. I will implement 'equals' when I get back home on Monday.

Reply:
The solution with records proposed by @holgerfriedlich already has the equals method defined by default if I am not mistaken. It is the same number of lines, so you could use that to avoid having to implement equals.

Reply:
I will use @holgerfriedrich 's record (with its correct implicit equals) to replace the class (with its implicit equals needing to be overriden) combined with my code to filter out all null records